### PR TITLE
Feature/adding calculator logic

### PR DESCRIPTION
**Pull request by angarita-dev**
### Introduced changes:

* Added calculate module
* Added operate module

@codingAngarita :wave:

### CODE REVIEW

File: src/logic/operate.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Big from 'big.js';
+
+export default function operate(numberOne, numberTwo, operation) {
+  const operations = {
+    '+': 'add',
+    '-': 'minus',
+    '÷': 'div',
+    '×': 'times',
+  };
```
Comment (suretrust) [inline src/logic/operate.js:9]: We should also take care of `%` in this file. 
* [ ] `operate` switches `operation` and uses `plus`, `minus`, `times`, `div` and `modifier` (turning the number into a percentage based number, `e.g. 95 % == .95`) functions ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone4/performOperate.png))

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi @codingAngarita, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [ ] `calculate.js` should be imported in the `App` component.

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

File: src/logic/calculate.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import operate from './operate';
+
+export default function calculate(calculatorData, buttonName) {
+  const numbers = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  const operations = ['+', '-', '÷', '×'];
+  const modifiers = ['+/-', '%'];
```
Comment (angarita-dev) [inline src/logic/calculate.js:6]: Here I added `+/-` and `%` as modifiers

File: src/logic/calculate.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+      next = addPoint(next);
+    }
+
+    return { total, next, operation };
+  }
+
+  // Applies modifier to current number
+  if (isModifier && (total || next)) {
+    const reverseSign = number => operate(number, -1, '×');
+    const makePercentage = number => operate(number, 100, '÷');
```
Comment (angarita-dev) [inline src/logic/calculate.js:77]: Here I used the operate module if the `buttonName` equals `%`.

**Comment by angarita-dev** [comment]
# Hi!

I currently treat the `%` logic in the `calculate` module. I implemented it as a "modifier" just as the `+/-`.
On the comments I highlighted were I addressed the modifiers.

As I implemented it the `operate` module has a single responsibility, it just operates two numbers, I believe that  adding the modifiers logic to the `operate` module would be a bad design design decision.

@codingAngarita :wave:

**Review by suretrust — APPROVED** [review]
Hi @codingAngarita, 👋 

### Even though this ☝️ is a requirement, I agree with your assessment/initiative. 🙌 Let's get this merged! 👍 

_Good luck with your future milestones and happy coding!_ 🎉

**Pull request merged**